packages/repository/src/relations/belongs-to/belongs-to.helpers.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright IBM Corp. 2018,2019. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import * as debugFactory from 'debug';
+import {InvalidRelationError} from '../../errors';
+import {isTypeResolver} from '../../type-resolver';
+import {BelongsToDefinition, RelationType} from '../relation.types';
+
+const debug = debugFactory('loopback:repository:belongs-to-helpers');
+
+/**
+ * @internal
+ */
+export type BelongsToResolvedDefinition = BelongsToDefinition & {keyTo: string};
+
+/**
+ * Resolves given belongsTo metadata if target is specified to be a resolver.
+ * Mainly used to infer what the `keyTo` property should be from the target's
+ * property id metadata
+ * @param relationMeta - belongsTo metadata to resolve
+ * @internal
+ */
+export function resolveBelongsToMetadata(relationMeta: BelongsToDefinition) {
+  if ((relationMeta.type as RelationType) !== RelationType.belongsTo) {
+    const reason = 'relation type must be BelongsTo';
+    throw new InvalidRelationError(reason, relationMeta);
+  }
+
+  if (!isTypeResolver(relationMeta.target)) {
+    const reason = 'target must be a type resolver';
+    throw new InvalidRelationError(reason, relationMeta);
+  }
+
+  if (!relationMeta.keyFrom) {
+    const reason = 'keyFrom is required';
+    throw new InvalidRelationError(reason, relationMeta);
+  }
+
+  const sourceModel = relationMeta.source;
+  if (!sourceModel || !sourceModel.modelName) {
+    const reason = 'source model must be defined';
+    throw new InvalidRelationError(reason, relationMeta);
+  }
+
+  const targetModel = relationMeta.target();
+  const targetName = targetModel.modelName;
+  debug('Resolved model %s from given metadata: %o', targetName, targetModel);
+
+  const targetProperties = targetModel.definition.properties;
+  debug('relation metadata from %o: %o', targetName, targetProperties);
+
+  if (relationMeta.keyTo) {
```

Comment:
I know you're getting the content of this file from `packages/repository/src/relations/belongs-to/belongs-to-accessor.ts`, but in the latest version it includes [this commit](https://github.com/strongloop/loopback-next/commit/ce7538511034a3ab3f7a55d780b00c1c08557439) that adds a constraint here (same thing for `HasMany` and `HasOne`).

Reply:
Yeah, because of that commit, I am getting merge conflicts when trying to rebase. My intention here is just to move `resolveBelongsToMetadata` into a new file, I don't want to revert any changes made on the `master` branch.

Reply:
See https://github.com/strongloop/loopback-next/pull/3410